### PR TITLE
MWPW-116985 Multi-column sections

### PR DESCRIPTION
**Pull request by Sartxi**
* Ability to say how many columns a section should be (two up, three up, four up).
* Ability to declare internal section widths (10, 8, 6, etc.)
* Ability to spec spacing between blocks.

Resolves: [MWPW-116985](https://jira.corp.adobe.com/browse/MWPW-116985)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/sarchibeque/section-grids
- After: https://simple-section-ups--milo--adobecom.hlx.page/drafts/sarchibeque/section-grids

### CODE REVIEW

File: libs/blocks/section-metadata/section-metadata.css [hunk 1]
```diff
@@ -68,29 +68,55 @@
...
-.section.two-up .grid,
-.section.three-up .grid,
-.section.four-up .grid {
+.section.two-up,
+.section.three-up,
+.section.four-up,
+.section.five-up {
   display: grid;
   grid-template-columns: repeat(auto-fit, minmax(276px, 1fr));
+  gap: var(--spacing-m);
```
Comment (meganthecoder) [inline libs/blocks/section-metadata/section-metadata.css:85]: I know @auniverseaway said simplified but I don't see why we wouldn't include the gap options. This is something Rikio already requested

Comment (Sartxi) [reply libs/blocks/section-metadata/section-metadata.css:85]: @meganthecoder @auniverseaway in my initial stab at this I created support for all the different consonant sizes for gap and column-gap/row-gap overrides. I can add all of them back in, or a subset of them. Any preferences?

Comment (ryanmparrish) [reply libs/blocks/section-metadata/section-metadata.css:85]: I think just adding the gap spacing tokens options will suffice. The column-gap and row-gap are nice to have, but might be more then we need for now.

File: test/blocks/icon-block/icon-block.test.js [hunk 2]
```diff
@@ -11,8 +11,8 @@ describe('icon blocks', () => {
   const blocks = document.querySelectorAll('.icon-block');
   blocks.forEach(block => {
     init(block);
-    const isVertical = block.classList.contains('vertical');
-    describe(`icon block ${isVertical ? "vertical" : "full-width"}`, () => {
+    const isColumn = block.classList.contains('vertical') || block.classList.contains('centered');
+    describe(`icon block ${isColumn ? "column" : "full-width"}`, () => {
```
Comment (ryanmparrish) [inline test/blocks/icon-block/icon-block.test.js:15]: nit: single quotes around 'column' : 'full-width'

File: libs/blocks/section-metadata/section-metadata.css (hunk 1 above)
Comment (Sartxi) [reply libs/blocks/section-metadata/section-metadata.css:117]: Okay I added back the section gap class tokens, we can add support for row or column if the use case arrises.

File: libs/blocks/icon-block/icon-block.js [hunk 3]
```diff
@@ -22,19 +22,19 @@ function decorateLayout(el) {
     decorateBlockBg(el, children[0]);
   }
   const foreground = document.createElement('div');
```
Comment (ryanmparrish) [inline libs/blocks/icon-block/icon-block.js:24]: Do we need a new element here? I think we can likely just use the given dom and add a 'foreground' class to the children[1] node if (children.length > 1). If a new el is warranted, we should import and use `createTag()`. 
Maybe a thought for future `icon-block` updates.

File: libs/blocks/icon-block/icon-block.css [hunk 4]
```diff
@@ -29,140 +29,140 @@
...
-  .icon-block.vertical .foreground.container.three-up .icon-area {
+  .three-up .icon-block.vertical .foreground .icon-area {
     max-width: 300px;
   }
 
-  .icon-block.vertical.four-up .foreground.container.four-up .icon-area {
+  .four-up .icon-block.vertical .foreground .icon-area {
     max-width: 276px;
   }
 }
```
Comment (ryanmparrish) [inline libs/blocks/icon-block/icon-block.css:168]: Nit: new line at end. Linter is barking about it.

File: libs/blocks/section-metadata/section-metadata.css [hunk 5]
```diff
@@ -68,29 +68,87 @@
...
+  .section.four-up {
     grid-template-columns: repeat(4, 1fr);
   }
 
-}
+  .section.five-up {
+    grid-template-columns: repeat(5, 1fr);
+  }
+
+}
```
Comment (ryanmparrish) [inline libs/blocks/section-metadata/section-metadata.css:154]: Nit: new line at end

**Review by ryanmparrish — CHANGES_REQUESTED** [review]

File: libs/blocks/section-metadata/section-metadata.css [hunk 6]
```diff
@@ -68,29 +68,87 @@
...
 .section.divider {
   border-bottom: 1px solid #d8d8d8;
 }
 
-.section.two-up .grid,
-.section.three-up .grid,
-.section.four-up .grid {
+.section.no-gap {
+  gap: 0;
+}
```
Comment (meganthecoder) [inline libs/blocks/section-metadata/section-metadata.css:81]: Will this work given you have a gap set for `.section.two-up` etc. Seems like either the gap styles need more specific selectors or you remove the gap set in the -ups

**Review by ryanmparrish — APPROVED** [review]

**Review by meganthecoder — APPROVED** [review]

**Review by auniverseaway — APPROVED** [review]

**Pull request merged**